src/ceph_syn.cc

```diff
@@ -88,16 +88,19 @@ int main(int argc, const char **argv, char *envp[])
     synclients.pop_front();
     syn->join_thread();
     delete syn;
+    syn = nullptr;
     delete client;
+    client = nullptr;
   }
 
   for (int i = 0; i < num_client; ++i) {
     // wait for messenger to finish
     delete mclients[i];
+    mclients[i] = nullptr;
     messengers[i]->shutdown();
     messengers[i]->wait();
     delete messengers[i];
+    messengers[i] = nullptr;
```

Comment:
i don't think it makes sense to reset the pointers before `return 0` in `main()`